🐍 Python Tutorial &nbsp;·&nbsp; *Bài 93/124*

[« Python Dates](092_python_dates.ipynb) &nbsp;•&nbsp; [📚 Mục lục](../README.md) &nbsp;•&nbsp; [Python JSON »](094_python_json.ipynb)

# Python Math

> 📖 **Học song song trên W3Schools:** [python_math.asp](https://www.w3schools.com/python/python_math.asp)
>
> 🎯 **Trong bài này:** Built-in Math Functions · ➕ Mở rộng: round(), sum(), divmod() · The Math Module

Python có sẵn một số **hàm toán học**, cùng với module **`math`** mở rộng giúp thực hiện các phép toán trên số.

## Built-in Math Functions — Hàm toán học có sẵn

`min()` và `max()` tìm giá trị nhỏ nhất/lớn nhất trong một iterable:

In [1]:
x = min(5, 10, 25)
y = max(5, 10, 25)

print(x)
print(y)

5
25


`abs()` trả về giá trị tuyệt đối (luôn dương):

In [2]:
x = abs(-7.25)
print(x)

7.25


`pow(x, y)` trả về `x` mũ `y`:

In [3]:
x = pow(4, 3)
print(x)

64


## ➕ Mở rộng: round(), sum(), divmod()

In [4]:
print(round(3.14159, 2))
print(round(2.5), round(3.5))   # làm tròn "về số chẵn" (banker's rounding)
print(sum([1, 2, 3, 4]))
print(divmod(17, 5))            # (thương, dư)

3.14
2 4
10
(3, 2)


## The Math Module — Module math

Python có module `math` mở rộng danh sách hàm toán học. Cần import trước khi dùng:

In [5]:
import math

x = math.sqrt(64)
print(x)

8.0


`math.ceil()` làm tròn **lên**, `math.floor()` làm tròn **xuống** tới số nguyên gần nhất:

In [6]:
import math

x = math.ceil(1.4)
y = math.floor(1.4)

print(x)
print(y)

2
1


Hằng số `math.pi` trả về giá trị số PI (3.14…):

In [7]:
import math

x = math.pi
print(x)

3.141592653589793


Một vài hàm/hằng số khác:

In [8]:
import math

print(math.e)              # số e
print(math.factorial(5))   # 5!
print(math.gcd(12, 18))    # ước chung lớn nhất
print(math.log10(1000))
print(math.sin(math.pi / 2))

2.718281828459045
120
6
3.0
1.0


Danh sách đầy đủ: [Math Module Reference](../10_reference/math_module.md).

> 🧪 **Try it Yourself** — Tính diện tích hình tròn bán kính r, làm tròn 2 chữ số thập phân.

In [9]:
import math

r = 5
area = math.pi * r ** 2
print(round(area, 2))

78.54


---

## 🧠 Ghi nhớ nhanh

> - Có sẵn: `min() max() abs() pow() round() sum()`.
> - Module `math`: `sqrt()` căn bậc hai · `ceil()` làm tròn **lên** · `floor()` làm tròn **xuống** · `pi` · `e` · `factorial()`.
> - ⚠️ `round(2.5)` → `2` (làm tròn về số chẵn gần nhất).

---

## 📝 Exercise — Bài tập nhanh

**❓ Câu hỏi:** `math.ceil(1.2)` trả về gì?

- **A.** `1`
- **B.** `2`
- **C.** `1.2`
- **D.** `1.0`

<details>
<summary>👉 Xem đáp án</summary>

✅ **Đáp án: B** — `2`

</details>

---

[« Python Dates](092_python_dates.ipynb) &nbsp;•&nbsp; [📚 Mục lục](../README.md) &nbsp;•&nbsp; [Python JSON »](094_python_json.ipynb)